alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -92,14 +84,12 @@ essentials::IdentifierConstPtr BasicBehaviour::getOwnId() const
  */
 bool BasicBehaviour::stop()
 {
-    {
-        std::lock_guard<std::mutex> lck(_runLoopMutex);
-        setSignalState(SignalState::STOP);
-        setStopCalled(true);
-    }
-    if (_behaviour->isEventDriven()) {
-        _runCV.notify_all();
+    if (!isStopCalled()) {
```

Comment:
Don't need to check if (!isStopCalled()) . If a running plan instance called stop that means it must have called a start before.